### ISSUE TITLE
[Core Workflow] Remove option does not work with tree select node that has sub-options

### ISSUE THREAD

**Issue by YetAnotherGerrit** [zammad/zammad#4407]
### Used Zammad Version

5.3

### Environment

- Installation method: any
- Operating system (if you're unsure: `cat /etc/os-release` ): any
- Database + version:  any
- Elasticsearch version: any
- Browser + version: any

### Actual behaviour

When you remove an option (core workflow) from a tree_select or multi_tree_select and this option has sub-options, it is not removed. You'll need to remove all sub-options, too.

### Expected behaviour

When you remove an option (core workflow) from a tree_select or multi_tree_select that has sub-options, it is removed including all sub-options.
If you specifically want to disable this option, without removing the sub-options a new action should be introduced "disable option".

### Steps to reproduce the behaviour

Create a tree_select for example:
A
- BB
  - CCC
  - DDD

Now create a workflow that removes node "BB". It will still show. Only if you include "CCC" and "DDD" in the workflow, "BB" will be removed.

### Support Ticket

Ticket#10111842

### I'm sure this is a bug and no feature request or a general question.

yes

**Comment by jouvin** [comment]
I raised an issue quite similar on the [forum](https://community.zammad.org/t/how-to-add-a-tree-select-parent-and-children-to-a-core-workflow/10868). The exact use case is the opposite: having an attribute like the one proposed, if you add to your core workflow that this attribute is `fixed` to `BB`, `CCC`  and `DDD` will not be displayed, despite the fact the `BB` as a `>` on its right side suggesting that it has children.

This use case and the original one are probably both related to the way tree are processed by the core workflow.

**Comment by dominikklein** [comment]
@YetAnotherGerrit But is it really a bug that BB is not removed? For me, it should be "disabled" for the selection, but the childrens should be available.

**Comment by YetAnotherGerrit** [comment]
> @YetAnotherGerrit But is it really a bug that BB is not removed? For me, it should be "disabled" for the selection, but the children should be available.

That is an interesting take on this subject and would actually solve another huge problem, a lot of my clients have: being able to force their users to select an end node of the tree.

So maybe we need to have two variants of the "remove option" for (multi) tree selects:
- disable this option, but keep the sub-options still accessible
- remove the option and all sub-options

The same goes for jouvin's approach:
- set fixed to a selection of options
- set fixed to a selection of options and their sub-options

But never the less. Currently, when you remove option BB. It does not have any effect at all. So I consider this still a bug. Because it's neither removing BB with its sub-options nor does it disable BB.

**Comment by jouvin** [comment]
I agree with @YetAnotherGerrit that ideally both approach should be possible. But if only one is implemented in the short term, I think that the current one is not appropriate as it defeats the usability of suboptions in core workflows (except by paying the price of an unmaintanaible configuration where you list all sub-options explicitely, may mean a list of tens if not hundred of entries...).

**Comment by dominikklein** [comment]
@rolfschmidt What do you think here?

**Comment by rolfschmidt** [comment]
> @rolfschmidt What do you think here?

Hard to say... Because of the weird behaviours of the fields (parent selectable or not) it is very hard to define a behaviour here. Disabling seems to be the way to go with the lowest impact.

**Comment by YetAnotherGerrit** [comment]
How was it solved now?

**Comment by rolfschmidt** [comment]
> How was it solved now?

We are disabling nodes which are excluded via core workflow but relevant for the selection to work.

**Comment by YetAnotherGerrit** [comment]
> > How was it solved now?
> 
> We are disabling nodes which are excluded via core workflow but relevant for the selection to work.

So the sub-options are still available, even though I disabled a node?

**Comment by rolfschmidt** [comment]
> > > How was it solved now?
> > 
> > 
> > We are disabling nodes which are excluded via core workflow but relevant for the selection to work.
> 
> So the sub-options are still available, even though I disabled a node?

Based on this example:

```
A
-> BB
--> CCC
--> DDD
```

If you remove BB then BB is disabled and CCC and DDD are still accessible.

If you remove BB+CCC+DDD then this part of the tree is gone.

If you remove CCC, then just CCC is removed.

If you remove A, everything is deleted.

**Comment by dominikklein** [comment]
> If you remove A, everything is deleted.

But I think in this situation `A` is disabled and the others are available, like for the first example.

**Comment by rolfschmidt** [comment]
> > If you remove A, everything is deleted.
> 
> But I think in this situation `A` is disabled and the others are available, like for the first example.

upsi yes

**Comment by YetAnotherGerrit** [comment]
This solution does not solve my issue and the original use-case. It may have fixed "a" bug, but not the issue.

**Comment by dominikklein** [comment]
@YetAnotherGerrit, kindly create a new issue if you believe there's a missing element. I believe we've outlined a viable solution for the problem described here.
At present, the design only allows for the removal of individual options, not an entire subtree (and this option will be shown disabled when it has available child nodes).
It appears we neglected to update the initial article with the refined solution.

Based on your current concerns, it seems to me that this is more akin to a feature request.
